### PR TITLE
Move version check code out of class Updater

### PR DESCRIPTION
**Pull request by DeepDiver1975**
This is a prestep before we can enhance the updater code - the version check is moved out

@LukasReschke @MorrisJobke please review - THX

### CODE REVIEW

File: apps/updatenotification/lib/updatechecker.php [hunk 1]
```diff
@@ -30,7 +30,7 @@ class UpdateChecker {
 	/**
 	 * @param Updater $updater
 	 */
-	public function __construct(Updater $updater) {
+	public function __construct(Updater\VersionCheck $updater) {
 		$this->updater = $updater;
```
Comment (MorrisJobke) [inline apps/updatenotification/lib/updatechecker.php]: seems like $updater of type object<OC\Updater\VersionCheck> is incompatible with the declared type object<OC\Updater> of property $updater.

**Comment by DeepDiver1975** [comment]
@LukasReschke I guess we might even want to think about moving the version check completely into the updatenotifications app

File: tests/lib/updater/versioncheck.php [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+	 * @return string
+	 */
+	private function buildUpdateUrl($baseUrl) {
+		return $baseUrl . '?version='.implode('x', Util::getVersion()).'xinstalledatxlastupdatedatx'.\OC_Util::getChannel().'x'.\OC_Util::getEditionString().'x';
+	}
+
+	/**
+	 * @return array
+	 */
+	public function versionCompatibilityTestData() {
```
Comment (ChristophWurst) [inline tests/lib/updater/versioncheck.php]: that method/data is not used?

File: lib/private/updater/versioncheck.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+use OC\Hooks\BasicEmitter;
+use OC\HTTPHelper;
+use OC_Util;
+use OCP\IConfig;
+use OC\Setup;
+use OCP\Util;
+
+class VersionCheck extends BasicEmitter {
+
+	/** @var \OC\HTTPHelper $helper */
```
Comment (ChristophWurst) [inline lib/private/updater/versioncheck.php]: HTTPHelper is [deprecated](https://github.com/owncloud/core/blob/master/lib/private/httphelper.php), you could use the IClientService instead

Comment (DeepDiver1975) [inline lib/private/updater/versioncheck.php]: I know - but I only want to move the code fragments around for now - no need to reimplement that all ... at least not now :wink:

File: lib/private/updater/versioncheck.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+namespace OC\Updater;
+
+use OC\Hooks\BasicEmitter;
+use OC\HTTPHelper;
+use OC_Util;
+use OCP\IConfig;
+use OC\Setup;
+use OCP\Util;
+
+class VersionCheck extends BasicEmitter {
```
Comment (nickvergessen) [inline lib/private/updater/versioncheck.php]: Emitter is not used?

**Comment by LukasReschke** [comment]
👍

**Comment by DeepDiver1975** [comment]
@nickvergessen commants addressed
@ChristophWurst deprecated HTTPHelper replaced with IClientService

Let's move this in - THX

**Comment by ChristophWurst** [comment]
:+1:

**Pull request merged**